onward/app/weather/controllers/WeatherController.scala

```diff
@@ -32,4 +54,61 @@ class WeatherController(weatherApi: WeatherApi, val controllerComponents: Contro
           Cached(10.minutes)(JsonComponent(views.html.weatherFragments.cityForecast(response)))
         })
     }
+
+  private def readLocationHeaders(implicit request: Request[AnyContent]) = {
+    val CityHeader = "X-GU-GeoCity"
+    val RegionHeader = "X-GU-GeoRegion"
+    val CountryHeader = "X-GU-GeoCountry"
+
+    def getEncodedHeader(key: String)(implicit request: Request[AnyContent]) =
+      request.headers.get(key).map(java.net.URLDecoder.decode(_, "latin1"))
+
+    val country = getEncodedHeader(CountryHeader)
+    val city = getEncodedHeader(CityHeader)
+    val region = getEncodedHeader(RegionHeader)
+
+    (country, city, region)
+  }
+
+  private def whatIsMyCity(maybeCountry: Option[String], maybeCity: Option[String], maybeRegion: Option[String])(
+      implicit request: Request[AnyContent],
+  ): Future[CityResponse] = {
+    (maybeCountry, maybeCity) match {
+      case (Some(countryCode), Some(city)) if countryCode.nonEmpty && city.nonEmpty =>
```

Comment:
Do we expect the string to be empty on occasion? If so, do we know why/when?

Reply:
And does it make a difference as the search would then just be for emptyString which would fail. 

Just asking as it seems to be guarding against something I can't quite work out?